README
                   .__       _________            .___                   
    .__       ____ |__|__  __\_   ___ \  ____   __| _/___________  ______
  __|  |___  /    \|  \  \/  /    \  \/ /  _ \ / __ |/ __ \_  __ \/  ___/
 /__    __/ |   |  \  |>    <\     \___(  <_> ) /_/ \  ___/|  | \/\___ \ 
    |__|    |___|  /__/__/\_ \\______  /\____/\____ |\___  >__|  /____  >
                 \/         \/       \/            \/    \/           \/ 
					* http://www.nixCoders.org/ *


1) Environnement vars
2) In game commands
3) Makefile options
4) Dependencies
5) Other features
6) FAQ

1) Environnement vars
----------------------
You can find and modify this vars in run.sh

ETH_CONF_FILE
	Config filename. Default is ~/.ethconf
ETH_IRC_NICK
	IRC nick. Default is $USER
ETH_LOG_FILE
	Log file use for verbose debug. Only works with debug build.
CLGUID
	For fake ET GUID
GUID
	For fake etpro GUID

2) In game commands
----------------------

eth_find
	Find all others eth users on server and add them as friend in the players list menu.
eth_invite
	Invite in your fireteam all friends in the players list menu.
eth_randomkick
	Vote kick player randomly.

eth_cleanpk3
	You can use this when you have 'Invalid pk3 error'. It remove all small pk3.
eth_exec [COMMAND]
	Execute a system command.

eth_spam
	Spam with 'eth powaa'.
eth_sysinfo
	Spam with system infos.
eth_save
	Saves the current ETH config (menu options) to a file. For example:
	\eth_save .etproconf if you wanna save config for playing on etpro servers
eth_load
	Same as eth_save, but this one loads configs. For example:
	\eth_load .etproconf when you wanna load config for playing on etpro
eth_irc_send
	If connected to IRC, it'll send a message to the bot channel
eth_irc_chat
	Same as eth_irc_send, but this one popups a window for writing a msg,
	same as the window for global/fireteam/team chat. You can bind to a key
	e.g. "bind X eth_irc_chat"
eth_showbuddies
	When IRC bot is connected, you can get list of other ETH users in console and see their ID and
	server where they play
eth_joinbuddy
	Join the server where a buddy is playing, usage:
	\eth_joinbuddy ID
eth_invitebuddy
	Invite a buddy to join you on the server, usage:
	eth_invitebuddy ID


3) Makefile options
----------------------

make
	Compile in debug mode. The binary is not optimized.

make RELEASE=1
	Compile for release version. The binary is optimized and stripped.
	
make dist
	Create a package for releasing. This one have sources, a compiled version,
	a ghf package and the needed files of sdk.

4) Dependencies
----------------------

Library use in eth.

- binutils 
- cppunit
- libx11
- libelf

5) Other Features
----------------------

Player list
	When "unmarked" category is opened, mouse1 will mark selected player as a friend
	and mouse2 will mark it as a teamkiller
	When "friends" or "teamkillers" category is opened, mouse1 will unmark selected player
	and mouse2 will mark him with the opposite of what it is at that moment (tk>friend, friend>tk)

6) FAQ
----------------------

I can't compile eth
- Read installation instructions in this manual and check requirements. If you're still
  unable to compile the bot yourself - use precombiled libETH.so and eth.pk3

I can't open the console, why?
- I don't know why, but i've noticed this on some distros. If the standard "~" key doesen't
  work, you'll have to toggle your console by pressing ALTGR + 1. I don't know if there is
  a solution to get this to normal. In case there is - notify us and we'll post it here.

My global/team/fireteam popup doesn't work!
- You've probably forgot to copy eth.pk3 to your ~/.etwolf/etmain/ folder. eth is using a
  modified chat window which is inside eth.pk3 and won't load the one located in game's pk3 files.

Medicbot is doing crazy things
- It's still in an early test version, don't be surprised if you find a bug in it

Punkbuster is kicking me for eth_ cvars!
- eth >= 1.0 doesn't use cvars. You've probably used the old eth which inserted cvars into your config file(s). You can edit your config files by hand and remove all eth_ cvars or as a faster solution you can delete all configs from your ~/.etwolf/


CHANGELOG
v1.1 10/10/2006
----------------------
Moved repo to SourceForge
Riflenade cam & bot
Backtrace 
Dlights
Hud fix

v1.1 22/09/2006
----------------------
Decided to update the CHANGELOG to make it appear as we pay attention to this file. - Sticky
ETPRO 3.2.6 Final Support
TCE Support -- Still a little buggy.

v1.0 28/08/2006
----------------------
Tracker Support via IRC status-bot.
Chat popup box for IRC.
Multi Mod Support
- ET 2.55/2.56/2.60/2.60b
- etpro 3.2.4/3.2.5/3.2.6-beta1
- etpub client 20051128/20060205/20060304/20060310/20060606
- jaymod 1.4/1.5-beta4/1.5-beta5/2.0.1
- whale 1.3/1.4-beta1

v0.95 12/05/2006
----------------------
IRC status-bot support - Optionally logs on to IRC with the other ETH users to show who's playing where.
IRC fireteam - Chat in IRC from ET with Global ETH users.
Supports Jaymod, etpubclient, whalemod, etpub, shrub, etmain.
ETH Hud added - For use with cg_draw2d 0.
Mortar tracers - shows where your mortar will be going, along with other mortar's (both axis & allies).
Mortar Cam / Hack - Displays where your mortar is going to land before you shoot.
Player radar - Shows where your teammates, and enemys are around you.
Added more chams.
Remove speclock option.
Improved sniper zoom removal.
Improved aimbot.
Panzer cam - Displays where you will be shooting the panzer.

v0.95-beta 13/02/2006
-----------------------
Implemented multi-mod support such as whale, etpubclient.

v0.6 - 10/11/2005
-----------------------
Aimbot (tag and vector based)
Players chams
Player name, dynamite, grenade, satchel esp
Dynamite counter
AutoDefense
Sniper zoom removal
Spawn timer
Auto action (aim at dynamite and dead player with the correct weapon)

v0.1 - beta - 09/08/2005
-----------------------
First release at elitecoders.org.
Cvar system.
Basic clienthook wallhack.


offsets.h
// GPL License - see http://opensource.org/licenses/gpl-license.php
// Copyright 2006 *nixCoders team - don't forget to credits us

/*
==============================
All offsets for mods/ET

Only offsets in this file
==============================
*/

#ifndef HOOK_H_
#define HOOK_H_

const static ethMod_t ethMods[] = {
	// No stripped binary mods	(only here to put a name/version on a mod)
	// Only put official mod infos there.
	{ "etmain",		"2.55",			0x1ae3ee02,	MOD_ETMAIN },
	{ "etmain",		"2.56",			0x4153a646,	MOD_ETMAIN },
	{ "etmain",		"2.60",			0xcb26a1fd,	MOD_ETMAIN },
	{ "etpub",		"20051128",		0x3f93ed44,	MOD_ETMAIN },
	{ "etpub",		"20051230",		0x4c8cb6a2,	MOD_ETMAIN },
	{ "etpub",		"20060205",		0x1e82f5d2,	MOD_ETMAIN },
	{ "etpub",		"20060304",		0x79c4963a,	MOD_ETMAIN },
	{ "etpub",		"20060310",		0x7077b613,	MOD_ETMAIN },
	{ "etpub",		"20060606",		0x95520381,	MOD_ETMAIN },
	{ "etpub",		"20060810",		0x59d32ce5,	MOD_ETMAIN },
	{ "etpub",		"20060818",		0x2e7903b6,	MOD_ETMAIN },
	{ "jaymod",		"1.4",			0xbd9bb6ad,	MOD_ETMAIN },
	{ "noquarter",	"1.0.1",		0x279e6121,	MOD_ETMAIN },
	{ "tce",		"0.48",			0x866bdb5b,	MOD_TCE },
	{ "tce",		"0.49",			0x965ad597,	MOD_TCE },
	{ "whale",		"1.3",			0x09285202,	MOD_ETMAIN },
	{ "whale",		"1.4-beta1",	0xe62c638e,	MOD_ETMAIN },

	// Stripped binary mods
	{ "etpro",	"3.2.4", 0x6a96361f, MOD_ETPRO,
		0x453b360, 2700,	// cg_entities
		0x75ef0,	// CG_Trace()
		0xa8940,	// BG_EvaluateTrajectory()
		0x4a0f0,	// CG_EntityEvent()
		0x964b0,	// CG_FinishWeaponChange()
		0x738f0,	// CG_DamageFeedback()
		0x92ab0,	// CG_RailTrail2()
		0x982a0,	// CG_WeaponFireRecoil()
		// etpro
		0xa100,		// guid_function
		0xf368		// cvar proof
	},
	{ "etpro",	"3.2.5", 0xc665c831, MOD_ETPRO,
		0x453b360, 2700,	// cg_entities
		0x75ef0,	// CG_Trace()
		0xa8940,	// BG_EvaluateTrajectory()
		0x4a0f0,	// CG_EntityEvent()
		0x964b0,	// CG_FinishWeaponChange()
		0x738f0,	// CG_DamageFeedback()
		0x92ab0,	// CG_RailTrail2()
		0x982a0,	// CG_WeaponFireRecoil()
		// etpro
		0xa100,		// guid_function
		0xf368		// cvar proof
	},
	{ "etpro",	"3.2.6-beta1", 0x2726f60c, MOD_ETPRO,
		0x453c6a0, 2700,	// cg_entities
		0x762f0,	// CG_Trace()
		0xa8d70,	// BG_EvaluateTrajectory()
		0x4a4b0,	// CG_EntityEvent()
		0x968b0,	// CG_FinishWeaponChange()
		0x73cb0,	// CG_DamageFeedback()
		0x92eb0,	// CG_RailTrail2()
		0x986d0,	// CG_WeaponFireRecoil()
		// etpro
		0xa360,		// guid_function
		0xf718		// cvar proof
	},
	{ "etpro",	"3.2.6-beta2", 0x594c2a3c, MOD_ETPRO,
		0x453c6a0, 2700,	// cg_entities
		0x76300,	// CG_Trace()
		0xa8d80,	// BG_EvaluateTrajectory()
		0x4a4c0,	// CG_EntityEvent()
		0x968c0,	// CG_FinishWeaponChange()
		0x73cc0,	// CG_DamageFeedback()
		0x92ec0,	// CG_RailTrail2()
		0x986e0,	// CG_WeaponFireRecoil()
		// etpro
		0xa360,		// guid_function
		0xf718		// cvar proof
	},
	{ "etpro",	"3.2.6", 0xc10e7d8c, MOD_ETPRO,
		0x453c6a0, 2700,	// cg_entities
		0x76300,	// CG_Trace()
		0xa8d80,	// BG_EvaluateTrajectory()
		0x4a4c0,	// CG_EntityEvent()
		0x968c0,	// CG_FinishWeaponChange()
		0x73cc0,	// CG_DamageFeedback()
		0x92ec0,	// CG_RailTrail2()
		0x986e0,	// CG_WeaponFireRecoil()
		// etpro
		0xa360,		// guid_function
		0xf718		// cvar proof
	},
	{ "jaymod",	"1.5-beta4", 0xb72cc38e, MOD_ETMAIN,
		0x29407c0, 2704,	// cg_entities
		0x6c7c0,	// CG_Trace()
		0xa8fa0,	// BG_EvaluateTrajectory()
		0x37700,	// CG_EntityEvent()
		0x91a90,	// CG_FinishWeaponChange()
		0x694b0,	// CG_DamageFeedback()
		0x8cd80,	// CG_RailTrail2()
		0x95180,	// CG_WeaponFireRecoil()
	},
	{ "jaymod",	"1.5-beta5", 0x3ffcf1a4, MOD_ETMAIN,
		0x29407c0, 2704,	// cg_entities
		0x6c7c0,	// CG_Trace()
		0xa8fa0,	// BG_EvaluateTrajectory()
		0x37700,	// CG_EntityEvent()
		0x91a90,	// CG_FinishWeaponChange()
		0x694b0,	// CG_DamageFeedback()
		0x8cd80,	// CG_RailTrail2()
		0x95180,	// CG_WeaponFireRecoil()
	},
	{ "jaymod",	"2.0.1", 0xc2051741, MOD_ETMAIN,
		0x28d6180, 2704,	// cg_entities
		0x692f0,	// CG_Trace()
		0xa2b70,	// BG_EvaluateTrajectory()
		0x3c360,	// CG_EntityEvent()
		0x8c450,	// CG_FinishWeaponChange()
		0x66170,	// CG_DamageFeedback()
		0x87c40,	// CG_RailTrail2()
		0x8f950,	// CG_WeaponFireRecoil()
	},
	{ "jaymod",	"2.0.2", 0x6528a7e0, MOD_ETMAIN,
		0x28d6180, 2704,	// cg_entities
		0x692f0,	// CG_Trace()
		0xa2b70,	// BG_EvaluateTrajectory()
		0x3c360,	// CG_EntityEvent()
		0x8c450,	// CG_FinishWeaponChange()
		0x66170,	// CG_DamageFeedback()
		0x87c40,	// CG_RailTrail2()
		0x8f950,	// CG_WeaponFireRecoil()
	},
	{ "jaymod",	"2.0.3", 0x15ba6692, MOD_ETMAIN,
		0x28d6180, 2704,	// cg_entities
		0x692f0,	// CG_Trace()
		0xa2b70,	// BG_EvaluateTrajectory()
		0x3c360,	// CG_EntityEvent()
		0x8c450,	// CG_FinishWeaponChange()
		0x66170,	// CG_DamageFeedback()
		0x87c40,	// CG_RailTrail2()
		0x8f950,	// CG_WeaponFireRecoil()
	}
};

// centity_t structs offsets
#define CG_ENTITIES_CURRENTSTATE_OFFSET 0x0
#define CG_ENTITIES_CURRENTVALID_OFFSET 0x244

/*
==============================
et.x86 addresses
==============================
*/

const static ethET_t ethET[] = {
	{	"2.55",		// version
		0x21e60afb,	// crc32
		0x080bae24, // Cvar_Set2
		0x080b561c,	// Cmd_AddCommand
		0x0804cbb4,	// COM_StripExtension
		0x080bee9c, // FS_ReadFile
		0x080c1d68,	// FS_PureServerSetLoadedPaks
		0x080b4e2c,	// Cbuf_ExecuteText
		0x087cd9c8,	// fs_searchpaths
		0x08f451d0,	// serverIP
		0x08ffc068,	// cl_mouseDx
		0x08ffc06c,	// cl_mouseDy
		0x0,		// etpro Linux checksum
		0x0,		// etpro win32 checksum
		#ifdef ETH_PRIVATE
			PRIV_ET255
		#endif
	},
	{	"2.56",		// version
		0x3d59a703,	// crc32
		0x080ba89c, // Cvar_Set2
		0x080b5098,	// Cmd_AddCommand
		0x0804cc34,	// COM_StripExtension
		0x080be914, // FS_ReadFile
		0x080c17e0,	// FS_PureServerSetLoadedPaks
		0x080b48a8,	// Cbuf_ExecuteText
		0x087cdfa8,	// fs_searchpaths
		0x08f457d0,	// serverIP
		0x08ffc668,	// cl_mouseDx
		0x08ffc66c,	// cl_mouseDy
		0x0,		// etpro Linux checksum
		0x0,		// etpro win32 checksum
		#ifdef ETH_PRIVATE
			PRIV_ET256
		#endif
	},
	{	"2.60",		// version
		0x3b18a889,	// crc32
		0x08070010, // Cvar_Set2
		0x08069240,	// Cmd_AddCommand
		0x08097880,	// COM_StripExtension
		0x080744e0, // FS_ReadFile
		0x08076570,	// FS_PureServerSetLoadedPaks
		0x08069650,	// Cbuf_ExecuteText
		0x0888c7c8,	// fs_searchpaths
		0x09036108,	// serverIP
		0x0906c1c8,	// cl_mouseDx
		0x0906c1cc,	// cl_mouseDy
		0xfe96e227,	// etpro Linux checksum
		0x6b04ecf8,	// etpro win32 checksum
		#ifdef ETH_PRIVATE
			PRIV_ET260
		#endif
	},
	{	"2.60b",	// version
		0x6ab49f82,	// crc32
		0x08073160, // Cvar_Set2
		0x0806b640,	// Cmd_AddCommand
		0x0,		// COM_StripExtension
		0x08079370, // FS_ReadFile
		0x0807cef0,	// FS_PureServerSetLoadedPaks
		0x0806bd70,	// Cbuf_ExecuteText
		0x088b1f18,	// fs_searchpaths
		0x090471c8,	// serverIP
		0x0907d288,	// cl_mouseDx
		0x0907d28c,	// cl_mouseDy
		0x395789a0,	// etpro Linux checksum
		0x30c3e725,	// etpro win32 checksum
		#ifdef ETH_PRIVATE
			PRIV_ET260B
		#endif
	}
};

#endif // HOOK_H_
